test/inductor/test_torchinductor.py

```diff
@@ -10558,6 +10559,88 @@ def fn(x):
         self.common(fn, (x,))
 
+    @skipCUDAIf(not SM80OrLater, "uses bfloat16 which requires SM >= 80")
+    @fresh_inductor_cache()
+    def test_dtypeview(self):
+        # https://github.com/pytorch/pytorch/issues/126338
+        def fn(x, y, x_dtype, x2):
+            x = x.view(x_dtype)
+            y = y.view(x_dtype) + 1
+            x2 = x2.view(x_dtype) + 1
+            return x @ y, x2 @ x
+
+        def _get_primitive_bitwidth(dtype):
+            if dtype.is_floating_point:
+                return torch.finfo(dtype).bits
+            else:
+                return torch.iinfo(dtype).bits
+
+        test_dtypes = [
+            torch.float32,
+            torch.float64,
+            torch.float16,
+            torch.bfloat16,
+            torch.uint8,
+            torch.int8,
+            torch.int16,
+            torch.int32,
+            torch.int64,
+        ]
+        for test_dtype_x in test_dtypes:
+            for test_dtype_y in test_dtypes:
+                # @ operation needs arguments to be the same dtype
+                for view_dtype in test_dtypes:
+                    if (
+                        _get_primitive_bitwidth(test_dtype_x)
+                        != _get_primitive_bitwidth(test_dtype_y)
+                        or _get_primitive_bitwidth(test_dtype_x)
+                        != _get_primitive_bitwidth(view_dtype)
+                        or view_dtype
+                        in [
+                            torch.int32,
+                            torch.int64,
+                            torch.int16,
+                            torch.uint8,
+                            torch.int8,
+                        ]
+                    ):
+                        continue
+                    # print(f"({test_dtype_x}, {test_dtype_y}, {view_dtype})")
+                    x = rand_strided(
+                        (2, 2), (2, 1), device=self.device, dtype=test_dtype_x
+                    )
+                    y = rand_strided(
+                        (2, 2), (2, 1), device=self.device, dtype=test_dtype_y
+                    )
+                    x2 = x.clone()
+                    self.common(
+                        fn,
+                        (x, y, view_dtype, x2),
+                        reference_in_float=False,
+                        check_lowp=False,
+                    )
+
+    def test_dtypeview_fusion(self):
+        @torch.compile
+        def fn(x):
+            x = x + 1
+            x = torch.ops.aten.view.dtype(x, torch.int16)
+            x = x * 2
+            return x
+
+        torch._inductor.metrics.generated_kernel_count = 0
+        x = torch.randn([1024], dtype=torch.float16, device=self.device)
+        self.common(fn, (x,), reference_in_float=False)
+        assertGeneratedKernelCountEqual(self, 1)
+
+    def test_reinterpret_dtypeview(self):
+        @torch.compile
+        def fn(x):
+            return x.view([10, 10]).view(torch.int32)
```

Comment:
can you also add the reverse (`view(torch.int32).view([10, 10])`) ? And check we didnt codegen ?

Reply:
For readers' context, `view(torch.int32).view([10,10])` in the previous implementation run without error but doesn't generate correct results which should be `reinterpret_tensor(aten.view.dtype())`.  It is because the ReinterpretView can't handle DtypeView input. This is fixed  inhttps://github.com/pytorch/pytorch/pull/128883/commits/a1996aaccefd0bb30f399ef5ae2d071e9d113731